### ISSUE TITLE
search_is_some can't be specifically allowed/denied

### ISSUE THREAD

**Issue by JohanLorenzo** [Manishearth/rust-clippy#820]
With this sample piece of code, the compiler fails [rustc 1.9.0-nightly (b678600ac 2016-03-29)].

```rust
#![feature(plugin)]
#![plugin(clippy)]
#![deny(clippy)]

#![allow(search_is_some)]

fn main() {
    let vec = vec![0, 1, 2];
    vec.iter().find(|&&x| x == 0).is_some();
}
```

Here's the error returned:
``` rust
src/lib.rs:9:5: 9:44 error: called `is_some()` after searching an iterator with find. This is more succinctly expressed by calling `any()`.
src/lib.rs:9     vec.iter().find(|&&x| x == 0).is_some();
                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/lib.rs:3:9: 3:15 note: lint level defined here
src/lib.rs:3 #![deny(clippy)]
                     ^~~~~~

src/lib.rs:5:10: 5:24 warning: unknown lint: `search_is_some`, #[warn(unknown_lints)] on by default
src/lib.rs:5 #![allow(search_is_some)]
```

I tried some other rules, like `similar_names`,  and I didn't notice any issue. I don't see what could be specific to `search_is_some`.

**Comment by llogiq** [comment]
This appears to be a problem in Rust, not in clippy (at least the code does follow lint levels). `rustc` sets the level for the `clippy` group to `deny` and apparently fails to `allow` it for the specific lint. I'll see if I can set up a test case that doesn't involve clippy and file a Rust issue.

This can stay open unless the issue is resolved.

**Comment by llogiq** [comment]
Scratch that, it's a bug in clippy. I'll have a fix up shortly.

**Comment by llogiq** [comment]
The problem was that the LintPass failed to register the lint with its `lint_array`.

**Comment by JohanLorenzo** [comment]
Thanks for the quick fix! :smiley:

**Comment by llogiq** [comment]
Thank you for reporting the error! :smile:
